server/service/osquery_utils/queries.go

```diff
@@ -173,24 +173,37 @@ var hostDetailQueries = map[string]DetailQuery{
 		},
 	},
 	"os_version_windows": {
+		// Fleet requires the DisplayVersion as well as the UBR (4th part of the version number) to
+		// correctly map OS vulnerabilities to hosts. The UBR is not available in the os_version table.
+		// The full version number is available in the `kernel_info` table, but there is a Win10 bug
+		// which is reporting an incorrect build number (3rd part), so we query the Windows registry for the UBR
+		// here instead.  To note, osquery 5.12.0 will have the UBR in the os_version table.
+
 		// display_version is not available in some versions of
 		// Windows (Server 2019). By including it using a JOIN it can
 		// return no rows and the query will still succeed
 		Query: `
-		WITH display_version_table AS (
-			SELECT data as display_version
-			FROM registry
-			WHERE path = 'HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\DisplayVersion'
-		)
-		SELECT
-			os.name,
-			COALESCE(d.display_version, '') AS display_version,
-			k.version
-		FROM
-			os_version os,
-			kernel_info k
-		LEFT JOIN
-			display_version_table d`,
+    WITH display_version_table AS (
+    	SELECT data as display_version
+    	FROM registry
+    	WHERE path = 'HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\DisplayVersion'
+    ),
+    ubr_table AS (
```

Comment:
@dantecatalfamo i'm not sure if UBR is consistently reported across all Windows version, but I followed the same pattern for DisplayVersion

Reply:
I can check, I have VMs running all versions of windows

Reply:
<img width="1931" alt="image" src="https://github.com/fleetdm/fleet/assets/43040593/20a78359-efd3-4653-936c-716f1f48e455">

Reply:
Unfortunately all of my hosts are up-to-date, so I can't test on old versions